### DISCUSSION TITLE
I made a simple store to share state between islands

### DISCUSSION THREAD

**Discussion by amoreaulemay** [freshframework/fresh#588] [Show and tell]
Although it is framework agnostic, I made a simple store solution that works well with Fresh to share state between islands.

I read previous discussions (#230 and #519) on the subject, and Zustand and Statery were suggested, but I could not get them to work how I needed them to do for my projects.

So I did a simple observer pattern that attaches to the window object. I tried to do a store with minimal dependencies, and the only one is a polyfill from corejs that auto imports if the browser doesn't support `structuredClone`.

It's now in a working state, and is documented well enough, I'm now working on adding more tests.

Here's the link to the repo : [https://github.com/amoreaulemay/fresh-store](https://github.com/amoreaulemay/fresh-store)

Hope it helps someone!

**Comment by rojvv** [comment]
Hell yeah, I will definitely need it.
